File: stacks_and_queues/queue/queue.py
class Queue:

    class Node:
        def __init__(self, value):
            self.value = value
            self.next = None

    def __init__(self):
        self.root = None
        self._size = 0

    def append(self, item: int) -> None:
        if not self.root:
            self.root = Queue.Node(item)
        else:
            pointer = self.root
            while pointer.next:
                pointer = pointer.next
            pointer.next = Queue.Node(item)
        self._size += 1

    def remove(self) -> int:
        if not self.root:
            raise ValueError("Empty Queue")

        temp = self.root.value
        self.root = self.root.next
        self._size -= 1

        return temp

    def __str__(self):
        if not self.root:
            return '[]'

        pointer = self.root
        txt = f'{pointer.value} '
        while pointer.next:
            pointer = pointer.next
            txt += f'{pointer.value} '
        return txt

    def __len__(self):
        return self._size

File: stacks_and_queues/queue/queue.cpp
#include <iostream>
#include <sstream>

using namespace std;


class Queue{

    private:
        class Node{
            public:
                int value; 
                Node* next; 
                
                Node(int value): value(value), next(nullptr) {}
        };

        Node* root = nullptr;
        unsigned int _size = 0;

    public:
        void append(int item){
            if(!(root)){
                root = new Node(item);
            }
            else{
                Node* pointer = root;
                while(pointer->next){
                    pointer = pointer->next;
                }
                pointer->next = new Node(item);
            }
            _size++;
        }

        int remove(){
            if(!(root)){
                throw runtime_error("Empty Queue");
            }
            int temp = root->value;
            root = root->next;
            _size--;
            return temp;
        }
    
        unsigned int size() const{
            return _size;
        }

        string toString() const {
            Node* pointer = root;
            stringstream ss;
            while (pointer) {
                ss << pointer->value;
                if (pointer->next) {
                    ss << " -> ";
                }
                pointer = pointer->next;
            }
            return ss.str();
    }
};
